src/openstudio_lib/ThermalZonesGridView.cpp

```diff
@@ -383,35 +383,32 @@ void ThermalZonesGridController::addColumns(const QString& /*category*/, std::ve
       });
 
       std::function<bool(model::ThermalZone*, const model::Schedule&)> setCoolingSchedule([](model::ThermalZone* z, model::Schedule t_s) {
+        bool result = false;
         if (boost::optional<model::ThermostatSetpointDualSetpoint> thermostat = z->thermostatSetpointDualSetpoint()) {
-          return thermostat->setCoolingSetpointTemperatureSchedule(t_s);
+          result = thermostat->setCoolingSetpointTemperatureSchedule(t_s);
         } else {
           model::ThermostatSetpointDualSetpoint t(z->model());
+          result = t.setCoolingSetpointTemperatureSchedule(t_s);
           z->setThermostatSetpointDualSetpoint(t);
-          return t.setCoolingSetpointTemperatureSchedule(t_s);
         }
+        return result;
       });
 
       boost::optional<std::function<void(model::ThermalZone*)>> resetCoolingSchedule([](model::ThermalZone* z) {
         if (boost::optional<model::ThermostatSetpointDualSetpoint> thermostat = z->thermostatSetpointDualSetpoint()) {
           thermostat->resetCoolingSetpointTemperatureSchedule();
+          if (!thermostat->heatingSetpointTemperatureSchedule()) {
+            z->resetThermostatSetpointDualSetpoint();
+          }
         }
       });
 
       boost::optional<std::function<bool(model::ThermalZone*)>> isDefaulted;
 
-      boost::optional<std::function<std::vector<model::ModelObject>(model::ThermalZone*)>> otherObjects([](model::ThermalZone* z) {
+      boost::optional<std::function<std::vector<model::ModelObject>(const model::ThermalZone*)>> otherObjects([](const model::ThermalZone* z) {
         std::vector<model::ModelObject> result;
-        boost::optional<model::Thermostat> thermostat = z->thermostat();
-        boost::optional<model::ThermostatSetpointDualSetpoint> thermostatDualSetpoint = z->thermostatSetpointDualSetpoint();
-        if (thermostat) {
+        if (boost::optional<model::ThermostatSetpointDualSetpoint> thermostat = z->thermostatSetpointDualSetpoint()) {
```

Comment:
True that at this point the only possibility of DualSetpoint (and Thermostat is an base class without a public Ctor, so effectively an abstract one)